safe/report/extractors/composer.py

```diff
@@ -240,29 +240,40 @@ def qgis_composer_extractor(impact_report, component_metadata):
 
     # Set default map to resize
 
-    # check show only impact
-    show_only_impact = setting('set_show_only_impact_on_report', False, bool)
-    layers = [impact_report.impact] + impact_report.extra_layers
-    layer_registry = QgsMapLayerRegistry.instance()
-    if not show_only_impact:
-        hazard_layer = layer_registry.mapLayers().get(
-            provenance['hazard_layer_id'], None)
-
-        aggregation_layer_id = provenance['aggregation_layer_id']
-        if aggregation_layer_id:
-            aggregation_layer = layer_registry.mapLayers().get(
-                aggregation_layer_id, None)
-            layers.append(aggregation_layer)
-
-        layers.append(hazard_layer)
-
-    # check hide exposure settings
-    hide_exposure_flag = setting('setHideExposureFlag', False, bool)
-    if not hide_exposure_flag:
-        # place exposure at the bottom
-        exposure_layer = layer_registry.mapLayers().get(
-            provenance['exposure_layer_id'])
-        layers.append(exposure_layer)
+    # Define the layers for the impact map.
+    if not impact_report.multi_exposure_impact_function:  # single IF
+        layers = [impact_report.impact] + impact_report.extra_layers
+    else:  # multi-exposure IF
+        layers = [] + impact_report.extra_layers
+
+    add_supplementary_layers = (
+        not impact_report.multi_exposure_impact_function or not (
+            impact_report.multi_exposure_impact_function.output_layers_ordered)
+    )
+    if add_supplementary_layers:
+        # Check show only impact.
+        show_only_impact = setting(
+            'set_show_only_impact_on_report', False, bool)
```

Comment:
We have set the default value in the `default_settings.py`. So, you can remove the default value (False) here.